Writerside/topics/raptor-xx.md

```diff
@@ -1,3 +1,435 @@
-# RaptorXX
+# Simple C++ RAPTOR Implementation
 
-Implementation of the simple Raptor in C++ for performance comparison in benchmarking.
+This section discusses the design, implementation, and optimization of a public transport routing system, developed
+using modern C++20/C++23 standards. The project incorporates advanced data structures and algorithms to enable
+efficient route planning, with an emphasis on cross-platform compatibility and robust build configurations via CMake.
+
+The C++ implementation is based heavily on the [Simple Raptor](simple-raptor.md "Simple RAPTOR Implementation"), but
+extends it with optimizations and modern C++ features. In this section, we focus on the C++ implementation, detailing
+the key components and challenges faced during development.
+
+The C++ implementation comprises:
+
+1. A schedule reader.
+2. A module for converting GTFS data into the RAPTOR structure.
+3. A RAPTOR algorithm module.
+4. A Geometry helper module
+5. A wrapper of [spdlog](https://github.com/gabime/spdlog) for logging purpose
+
+```plantuml
+@startuml
+package "benchmark" {
+}
+
+package "test" {
+}
+
+package "src" {
+    package "schedule" {
+        [GtfsReader]
+    }
+    package "gtfsRaptorConfig" {
+        [GtfsToRaptorConverter]
+    }
+    package "raptor" {
+        [RaptorRouter]
+    }
+}
+
+[benchmark] --> [src] : uses
+[test] --> [src] : uses
+[GtfsReader] --> [GtfsToRaptorConverter] : provides data to
+[GtfsToRaptorConverter] --> [RaptorRouter] : converts to
+
+@enduml
+```
+
+We aimed to utilize the most modern C++ features possible. Although we initially attempted to implement the project
+using the new C++ module system, limited compiler support (primarily from MSVC on Windows) posed challenges.
+
+## Challenges of Implementing in C++ Compared to Java
+
+Implementing RaptorXX in C++ presented several challenges, particularly in comparison to Java. Below, we outline key
+areas where the two languages differ, highlighting the complexities faced during C++ development.
```

Comment:
General comment on this sections. I would avoid having the structure
C++: .....
Java: ....

Because most of the points you are making are: `Java (and other high level languages) natively support more utility methods, which C++ does not. To tackle this we did XY in C++.`

Reply:
Would you like to have the section removed completely?